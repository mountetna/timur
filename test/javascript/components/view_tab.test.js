import React from 'react';
import {render, screen, waitFor, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ViewTab from '../../../lib/client/jsx/components/browser/view_tab';

describe('ViewTab', () => {
  it('renders without a regex', async () => {
    const {asFragment} = render(
      <ViewTab
        tab={{
          name: 'default',
          title: 'Default',
          description: null,
          panes: [
            {
              name: 'pane1',
              items: [
                {
                  name: 'attribute1',
                  type: 'markdown',
                  text: 'This is a test'
                }
              ],
              title: 'Pane1'
            }
          ]
        }}
        record_name='PROJ1-HS1'
        model_name='Subject'
      />
    );

    await waitFor(() => screen.getByText('Pane1'));

    expect(screen.queryByText(/Pane1/)).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with a regex that matches the record_name', async () => {
    const {asFragment} = render(
      <ViewTab
        tab={{
          name: 'default',
          title: 'Default',
          description: null,
          panes: [
            {
              name: 'pane1',
              items: [
                {
                  name: 'attribute1',
                  type: 'markdown',
                  text: 'This is a test'
                }
              ],
              title: 'Pane1'
            }
          ],
          regex: 'PROJ1'
        }}
        record_name='PROJ1-HS1'
        model_name='Subject'
      />
    );

    await waitFor(() => screen.getByText('Pane1'));

    expect(screen.queryByText(/Pane1/)).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  it('does not render if regex false', () => {
    const {container} = render(
      <ViewTab
        tab={{
          name: 'default',
          title: 'Default',
          description: null,
          panes: [
            {
              name: 'pane1',
              items: [
                {
                  name: 'attribute1',
                  type: 'markdown',
                  text: 'This is a test'
                }
              ],
              title: 'Pane1'
            }
          ],
          regex: '.*SCG.*'
        }}
        record_name='PROJ1-HS1'
        model_name='Subject'
      />
    );
    expect(container.innerHTML).toHaveLength(0);
  });
});
