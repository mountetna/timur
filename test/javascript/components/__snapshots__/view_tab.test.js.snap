// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewTab renders with a regex that matches the record_name 1`] = `
<DocumentFragment>
  <div
    id="tab"
  >
    <div
      class="pane_group"
    >
      <div
        class="pane"
      >
        <div
          class="title"
          title="pane1"
        >
          Pane1
        </div>
        <div
          class="items"
        >
          <div
            class="view_item"
          >
            <div
              class="item_name"
            >
              attribute1
            </div>
            <div
              class="item_view"
            >
              <div
                class="item"
              >
                <div
                  class="markdown"
                >
                  <p>
                    This is a test
                  </p>
                  

                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ViewTab renders without a regex 1`] = `
<DocumentFragment>
  <div
    id="tab"
  >
    <div
      class="pane_group"
    >
      <div
        class="pane"
      >
        <div
          class="title"
          title="pane1"
        >
          Pane1
        </div>
        <div
          class="items"
        >
          <div
            class="view_item"
          >
            <div
              class="item_name"
            >
              attribute1
            </div>
            <div
              class="item_view"
            >
              <div
                class="item"
              >
                <div
                  class="markdown"
                >
                  <p>
                    This is a test
                  </p>
                  

                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
